src/App.vue

```diff
@@ -162,6 +164,52 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
     return (process.env.JEST_WORKER_ID !== undefined)
   }
 
+  /** Called when component is created. */
+  private created (): void {
+    // before unloading this page, if there are changes then prompt user
+    window.onbeforeunload = (event) => {
+      if (this.haveChanges) {
+        event.preventDefault()
+        // NB: custom text is not supported in all browsers
+        event.returnValue = 'You have unsaved changes. Are you sure you want to leave?'
+      }
+    }
+  }
+
+  /** Called to redirect to dashboard. */
+  private goToDashboard (): void {
```

Comment:
This method has to be here (instead of in Actions.vue) because the Confirm Dialog is on this page.